src/main/java/discord4j/discordjson/json/InteractionData.java

```diff
@@ -0,0 +1,44 @@
+package discord4j.discordjson.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import discord4j.discordjson.json.gateway.Dispatch;
+import discord4j.discordjson.possible.Possible;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableInteractionData.class)
+@JsonDeserialize(as = ImmutableInteractionData.class)
+public interface InteractionData extends Dispatch {
```

Comment:
As discussed, we need to split InteractionCreate from InteractionData.

The former can contain the latter using JsonUnwrapped.

Reply:
I somehow didn't push my changes, that's on me. That should fix this.